### DISCUSSION TITLE
How are Joomla custom libraries that already have a composer.json handled?

### DISCUSSION THREAD

**Discussion by alexandreelise** [joomlatools/joomlatools-pages-server#100] [Q&A] [answered]
Hi @johanjanssens,

I saw that you consolidate composer.json files with the merge composer plugin but does it merge also all composer.json file in a custom joomla library for example?
e.g ``` JPATH_ROOT/libraries/lib_example/composer.json ```
Please shed a light on that part.

**Comment by johanjanssens** [comment, marked as answer]
No. Composer was originally build to solve the problem of managing local dependencies per app or per site. 

Joomlatools Server uses composer to load required dependencies to run it's own internal dashboard. This is totally separate and not related to using composer in a Joomla site. You can run composer in a site if you wish to load dependencies, or not. Composer is globally installed but dependencies are locally managed.

**Reply by alexandreelise** [reply]
Thanks for clarifying. Happy coding Johan.
